### PR TITLE
Webapp Updates and Bug Fixes

### PR DESCRIPTION
**Pull request by tsteenbe**
See individual commits for details.

### CODE REVIEW

File: reporter-web-app/src/components/IssuesTable.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ class IssuesTable extends React.Component {
         } = this.props;
         const {
             filteredInfo = {},
-            sortedInfo
+            sortedInfo =  {}
```
Comment (fviernau) [inline reporter-web-app/src/components/IssuesTable.js:55]: commit message: 

support -> supports
React object -> React objects

maybe also:
sort enabled -> sorting enabled

File: reporter-web-app/src/components/RuleViolationsTable.js [hunk 2]
```diff
@@ -162,31 +162,37 @@ class RuleViolationsTable extends React.Component {
                 onFilter: (value, webAppRuleViolation) => {
                     const webAppPackage = webAppRuleViolation.package;
 
-                    if (value === 'excluded') {
-                        return webAppPackage.isExcluded;
-                    }
+                    if (webAppPackage) {
```
Comment (fviernau) [inline reporter-web-app/src/components/RuleViolationsTable.js:165]: Could this be simplified as follows? (maybe it's out of scope of this commit)
```
onFilter: (value, webAppRuleViolation) => {
  const webAppPackage = webAppRuleViolation.package;
  const isExcluded = webAppPackage && webAppPackage.isExcluded

  return value === 'excluded' ?: isExcluded : !isExcluded
}
```

or

```
onFilter: (value, webAppRuleViolation) => {
  if (!webAppRuleViolation.hasPackage) return false

  const isExcluded = webAppRuleViolation.package.isExcluded

  return isExcluded && value === 'excluded' || !isExcluded && value === 'included'
}
```

**Review by fviernau — CHANGES_REQUESTED** [review]

**Review by fviernau — DISMISSED** [review]

File: reporter-web-app/src/components/RuleViolationsTable.js (hunk 2 above)
Comment (fviernau) [reply reporter-web-app/src/components/RuleViolationsTable.js]: Has been implemented in a separate commit on top.

File: reporter-web-app/src/components/PackageDetails.js [hunk 3]
```diff
@@ -40,6 +40,18 @@ const PackageDetails = (props) => {
         vcsProcessed
     } = webAppPackage;
 
+    const {
```
Comment (sschuberth) [inline reporter-web-app/src/components/PackageDetails.js]: Commit message nits: s/reporter-webapp/reporter-web-app/

Can you also briefly explain how it was before, if not for each package?

File: reporter-web-app/src/components/IssuesTable.js [hunk 4]
```diff
@@ -48,12 +48,11 @@ class IssuesTable extends React.Component {
             issues,
             onChange,
             showExcludesColumn,
-            state
+            state: {
```
Comment (sschuberth) [inline reporter-web-app/src/components/IssuesTable.js:51]: Commit message nit: s/compacter/more compact/

File: reporter-web-app/src/components/PackageDetails.js [hunk 5]
```diff
@@ -124,6 +124,7 @@ const PackageDetails = (props) => {
             }
             {
                 !!vcsRevision
+                && vcsRevision !== vcsProcessedRevision
```
Comment (sschuberth) [inline reporter-web-app/src/components/PackageDetails.js:127]: Commit message nit: Double space in "display  package's"

**Review by fviernau — APPROVED** [review]

File: reporter-web-app/src/components/PackageDetails.js (hunk 3 above)
Comment (sschuberth) [reply reporter-web-app/src/components/PackageDetails.js]: I don't see this being addressed in the commit message.

File: reporter-web-app/src/App.css [hunk 6]
```diff
@@ -101,15 +101,15 @@ tr.ant-table-expanded-row, tr.ant-table-expanded-row:hover {
 }
 
 .ort-hint {
-    color: #FFCC00;
+    color: #84b6eb;
```
Comment (sschuberth) [inline reporter-web-app/src/App.css:104]: Nit: Existing code uses uppercase hex chars.

File: reporter-web-app/src/components/RuleViolationsTable.js [hunk 7]
```diff
@@ -160,19 +160,11 @@ class RuleViolationsTable extends React.Component {
                 filteredValue: filteredInfo.excludes || null,
                 key: 'excludes',
                 onFilter: (value, webAppRuleViolation) => {
-                    const webAppPackage = webAppRuleViolation.package;
-
-                    if (webAppPackage) {
-                        if (value === 'excluded') {
-                            return webAppPackage.isExcluded;
-                        }
+                    if (!webAppRuleViolation.hasPackage()) return true;
```
Comment (sschuberth) [inline reporter-web-app/src/components/RuleViolationsTable.js:163]: Explaining the purpose of the refactoring in the commit message would be nice.

**Review by sschuberth — APPROVED** [review]
Approving despite my comments to move things forward.

**Pull request merged**